src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -211,3 +215,78 @@ def _specific_row(self, slot, market):
                     self.area.strategy.panel_count
                     ]
         return []
+
+
+class KPI:
+    def __init__(self):
+        self.performance_index = dict()
+        self._cep_device = list()
+        self._cep_energy = 0
+        self._total_energy = 0
+
+    def __repr__(self):
+        return f"KPI: {self.performance_index}"
+
+    def _accumulated_trade_energy(self, area):
+        for child in area.children:
+            if isinstance(child.strategy, (CommercialStrategy, FinitePowerPlant)):
+                self._cep_device.append(child.name)
+                self._accumulate_energy(area, child, True)
+            elif _is_load_node(child) or _is_prosumer_node(child):
+                self._accumulate_energy(area, child, False)
+
+    def _accumulate_energy(self, area, child, is_cep: bool):
+        for markets in area.past_markets:
+            for trade in markets.trades:
+                if is_cep:
+                    if trade.offer.seller not in self._cep_device:
+                        continue
+                    self._cep_energy += trade.offer.energy
+                else:
+                    if trade.buyer is child.name:
+                        self._total_energy += trade.offer.energy
+
+    def _export_house_pv_self_consumption(self, area):
+        house_pv_device = list()
+        house_load_device = list()
+        trade_by_pv = 0
+        total_energy_bought = 0
+        traded_from_pv = 0
+        if not _is_house_node(area):
+            return
+        for child in area.children:
+            if isinstance(child.strategy, PVStrategy):
+                house_pv_device.append(child.name)
+            elif _is_load_node(child) or _is_prosumer_node(child):
+                house_load_device.append(child.name)
+
+        for markets in area.past_markets:
+            for trade in markets.trades:
+                # Total Electricity traded by house device
+                if trade.buyer in house_load_device:
+                    total_energy_bought += trade.offer.energy
+                # Electricity produced by PV
+                if trade.offer.seller in house_pv_device:
+                    trade_by_pv += trade.offer.energy
+                # PV electricity self_consumption
+                if trade.offer.seller in house_pv_device and trade.buyer in house_load_device:
+                    traded_from_pv += trade.offer.energy
+
+        if trade_by_pv != 0:
+            self_consumption_within_house = traded_from_pv / trade_by_pv
+        else:
+            self_consumption_within_house = 0
+        if total_energy_bought != 0:
+            self_sufficiency = traded_from_pv / total_energy_bought
+        else:
+            self_sufficiency = 0
+
+        return {"self_consumption_within_house": self_consumption_within_house,
+                "self_sufficiency": self_sufficiency}
+
+    def area_performance_index(self, area):
```

Comment:
Nitpicking, but this function would be better described if named 'update_kpis_from_area' (in order to be able to add more kpis in the future and not just the area_performance_index)